src/main/scala/com/microsoft/hyperspace/actions/RefreshActionBase.scala

```diff
@@ -64,13 +64,23 @@ private[actions] abstract class RefreshActionBase(
 
   // Reconstruct a df from schema
   protected lazy val df = {
-    val rels = previousIndexLogEntry.relations
-    val dataSchema = DataType.fromJson(rels.head.dataSchemaJson).asInstanceOf[StructType]
-    spark.read
-      .schema(dataSchema)
-      .format(rels.head.fileFormat)
-      .options(rels.head.options)
-      .load(rels.head.rootPaths: _*)
+    val rel = previousIndexLogEntry.relations.head
+    val dataSchema = DataType.fromJson(rel.dataSchemaJson).asInstanceOf[StructType]
+    if (rel.fileFormat.equals("delta")) {
+      // Exclude "versionAsOf" and "timestampAsOf" options, so that this DataFrame for refresh
+      // can see the latest snapshot.
+      spark.read
+        .schema(dataSchema)
+        .format(rel.fileFormat)
+        .options(rel.options - "versionAsOf" - "timestampAsOf")
+        .load(rel.rootPaths.head)
```

Comment:
you can probably use just this section because versionAsOf and timestampAsOf are only used in delta. the same code would work for all supported formats right?